DEPS

```diff
@@ -107,7 +107,7 @@ hooks = [
     'action': [
       'python',
       '-c',
-      'import os; os.chdir("src"); os.chdir("electron"); os.system("npm install")',
+      'import os, subprocess, sys; os.chdir(os.path.join("src", "electron")); npm = "npm.cmd" if sys.platform == "win32" else "npm"; subprocess.check_call([npm, "install"]);',
```

Comment:
how come `npm.cmd` is now needed on windows? afaik just `npm` was working fine previously.

If this is truly needed, then this has become too complicated for a one-liner & should be moved to a script.

Reply:
> `npm` was working fine previously

Only when you use `os.system()`.

`subprocess.call()` needs different executable names on different platforms, see here for example https://github.com/electron/electron/blob/52fe92d02ef94712fc7b19b3f03d3363cfa71a4d/script/lib/util.py#L28-L30

Reply:
I can move the `npm` call to a Python script.
You think it would be better to move the whole thing to a new script?

Reply:
IMO the `if` statement is enough to push it over the edge of "shouldn't be inlined in a string in a config file". The only reason these are in Python instead of just being direct commands is because there's no way to `cd` in a cross-platform way, and npm doesn't have a `-C` argument like `make` and `ninja`.

I'd be fine with having a `script/cd-npm.py` that took a directory as its first argument and passed the rest of its arguments directly to `npm` (or `npm.cmd`).

Reply:
@nornagon Any thoughts?